src/plugins/NEMD/MettDeamon.cpp

```diff
@@ -1888,7 +1795,7 @@ uint32_t Reservoir::getActualBinIndex() {return _binQueue->getActualBinIndex();}
 uint64_t Reservoir::getNumMoleculesLocal() {return _binQueue->getNumParticles();}
 uint32_t Reservoir::getNumBins() {return _binQueue->getNumBins();}
 std::vector<Molecule>& Reservoir::getParticlesActualBin() {return _binQueue->getParticlesActualBin();}
-bool Reservoir::nextBin(uint64_t& nMaxID) {bool bSuccess = _binQueue->next(); nMaxID += _density.at(0).numMolecules.global; return bSuccess;}
```

Comment:
@pramathe 
Please have a look here. It seems like there was no value assigned to `_density.at(0).numMolecules.global` before. I changed it to `_numMoleculesRead` as this variable contains the number of particles in the reservoir. I hope this is correct and represents `numMolecules.global`.

Reply:
Set in line 1856?

Reply:
Should be ok!